lib/resque/plugins/cloudwatch-monitor/cloudwatch-monitor.rb

```diff
@@ -43,19 +36,58 @@ def value
         1
       end
 
-      #Job on failure hook. receives the job arguments and the exception
+      #Job on on_failure hook. receives the job arguments and the exception
       def on_failure_report_cw(e, *args)
-        report_dimensions = dimensions(*args)
-        report(fail_namespace, metric_name, report_dimensions)
+        type = :on_failure
+        return unless should_report?(type)
```

Comment:
@vladshub why not extract the two shared lines in each function to one function and in each method call it with the appropriate type. for example ``report_foo(:on_failure)``